### DISCUSSION TITLE
Work towards KWin(FT) integration

### DISCUSSION THREAD

**Discussion by antis81** [lxqt/lxqt#2076] [General]
Hey fellow LXQt warriors!

Personally I am with a minimal KWin setup (strictly `lxqt + sddm + kwinft + systemsettings`. Setup works pretty well (espeicially helps win*doze users). LXQt footprint still is about ~500MiB - thus relatively light on RAM. 

@tsujan Do you think we can collaborate with KDE on certain integrational issues rather than "pushing away"? For example I could provide a merge request for KDE and help them fix their "transient" strategy (https://bugs.kde.org/show_bug.cgi?id=442044). This could be such a "game changer" opening the opportunity to better integrate (old and new) WM's don't you think?

**Comment by tsujan** [comment]
I'm not familiar with KDE codes.

**Reply by antis81** [reply]
Actually I had a look in KWin's codebase (Their style guide is really, really horrible! :-1: Willing to ignore this though and focus on getting the task done!).

Specifically the problem with KDE's "transient window" concept is:
It works on a per-process basis. Thus PCManFM is likely the only place where this behaviour becomes visible. It works as long as the desktop and file manager are each running in a dedicated process. Even Lumina (quite similar to LXQt, but Kernel agnostic) uses **two** processes (one for desktop - one for the FM (and subsequent windows, just like pcmanfm)).

As a result KDE does not see this as a bug. The only way I see to solve this is in a collaboration where LXQt steps in and supportsm with a proper KWin integration. (Since I know where to implement this I am willing to actually do it - if nobody else is interested.)

**Reply by tsujan** [reply]
> It works on a per-process basis

Yes, I know that because of its behavior. It *is* a bug because Desktop is a kind of window and there's no guarantee that its process should be different from non-Desktop windows.

**Reply by antis81** [reply]
Circles… :smiley_cat:

Reported this 1.5 weeks ago to KDE. Status **needs info**. Provided an example reproducing this plus complete information! Since then no further reply. What is going on?

**Reply by tsujan** [reply]
> What is going on?

It's normal. They may not have time.

**Reply by antis81** [reply]
Got quite positive feedback from KDE now having reproduced the issue. Aparently the "show desktop" behaviour is only reproducible on X11 and Wayland users may not be affected.
